domain/secret/watcher_test.go

```diff
@@ -279,6 +279,57 @@ func (s *watcherSuite) TestWatchObsoleteForUnitsOwned(c *gc.C) {
 	wCSingleUnit.AssertNoChange()
 }
 
+func (s *watcherSuite) TestWatchObsoleteForUserSecrets(c *gc.C) {
+	ctx := context.Background()
+	svc, st := s.setupServiceAndState(c)
+
+	data := coresecrets.SecretData{"foo": "bar", "hello": "world"}
+	uri1 := coresecrets.NewURI()
+	uri2 := coresecrets.NewURI()
+	c.Logf("uri1: %v, uri2: %v", uri1, uri2)
+
+	err := st.CreateUserSecret(ctx, 1, uri1, secret.UpsertSecretParams{
+		Data: data,
+	})
+	c.Assert(err, jc.ErrorIsNil)
+	err = st.CreateUserSecret(ctx, 1, uri2, secret.UpsertSecretParams{
+		Data:      data,
+		AutoPrune: ptr(true),
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	w, err := svc.WatchObsoleteUserSecretsToPrune(ctx)
+	c.Assert(err, gc.IsNil)
+	c.Assert(w, gc.NotNil)
+	defer workertest.CleanKill(c, w)
+
+	wc := watchertest.NewNotifyWatcherC(c, w)
+
+	// Wait for the initial changes.
+	wc.AssertOneChange()
+
+	// create revision 2, and obsolete revision 1.
```

Comment:
Can we comment here or split the checks - uri1 expects no events because it is not auto prune